### ISSUE TITLE
heftia-effects-0.4.0.0: Could not find module ‘GHC.TypeLits.KnownNat.Solver’

### ISSUE THREAD

**Issue by juhp** [sayo-hs/heftia#15]
The new version is failing somehow in Stackage Nightly:

```
       Preprocessing executable 'SemanticsZoo' for heftia-effects-0.4.0.0..                                                         
       Installing library in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea4451395ec473d03f
cbc958590d23b15706959684c77259fd/9.8.2/lib/x86_64-linux-ghc-9.8.2/heftia-effects-0.4.0.0-EIgn40AMK4RFNC17qMuOww                     
       Installing executable Writer in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea445139
5ec473d03fcbc958590d23b15706959684c77259fd/9.8.2/bin                                                                                
       Installing executable Teletype in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea4451
395ec473d03fcbc958590d23b15706959684c77259fd/9.8.2/bin                                                                              
       Installing executable Continuation in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea
4451395ec473d03fcbc958590d23b15706959684c77259fd/9.8.2/bin                                                                          
       Installing executable FileSystemProvider in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea4451395ec473d03fcbc958590d23b15706959684c77259fd/9.8.2/bin                                                                    
       Installing executable KeyedEffects in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea4451395ec473d03fcbc958590d23b15706959684c77259fd/9.8.2/bin                                                                          
       Installing executable Logging in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea4451395ec473d03fcbc958590d23b15706959684c77259fd/9.8.2/bin                                                                               
       Installing executable Continuation2 in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea4451395ec473d03fcbc958590d23b15706959684c77259fd/9.8.2/bin                                                                         
       Installing executable SemanticsZoo in /home/curators/work/unpack-dir/.stack-work/install/x86_64-linux-tinfo6/d7d205a89b6867ea4451395ec473d03fcbc958590d23b15706959684c77259fd/9.8.2/bin                                                                          
       Error: Cabal-simple_HwdwpEmb_3.10.2.0_ghc-9.8.2:                                                                             
       '/home/curators/.stack/programs/x86_64-linux/ghc-tinfo6-9.8.2/bin/ghc-9.8.2'                                                 
       exited with an error:                                                                                                        
       <command line>: Could not find module ‘GHC.TypeLits.KnownNat.Solver’.
```

We may have to add an upperbound or disable part of the build.

([also](https://github.com/commercialhaskell/stackage/commit/ebd06eab255b0afef13fa2bf987f4c4bd64efc13), unrelated, FYI eff (used by you benchmarks) is not on Hackage)

cc @danburton

**Comment by ymdryo** [comment]
Oh, I'm sorry for the trouble! I will check it.

**Comment by ymdryo** [comment]
The build works with Cabal, and version 0.7.12 of `ghc-typelits-knownnat` in nightly-2024-10-11 includes `GHC.TypeLits.KnownNat.Solver`, so I’m unable to identify the cause of the error. I want to reproduce that error on my end and am looking into how to build the latest commit of commercialhaskell/stackage.

If the executables seem to be the cause of the error, feel free to exclude them from the Stackage build. They are just example codes for using the library, not part of the library itself.

**Comment by ymdryo** [comment]
I have just uploaded the fixed version v0.4.0.1 to Hackage. Could you please check it?

**Comment by ymdryo** [comment]
I tried using nightly-2024-10-21, and it seems the issue has been resolved, so I will close this. Thank you for your work on Stackage! It was very helpful.

If anything comes up, please feel free to reopen.
